src/server/game/Scripting/ScriptDefines/PlayerScript.cpp

```diff
@@ -153,6 +153,14 @@ void ScriptMgr::OnPlayerTalentsReset(Player* player, bool noCost)
     });
 }
 
+void ScriptMgr::OnAfterSpecSlotChanged(Player* player, uint8 newSlot)
+{
+    ExecuteScript<PlayerScript>([&](PlayerScript* script)
```

Comment:
```suggestion
    ExecuteScript<PlayerScript>([=](PlayerScript* script)
```

Reply:
I'm fine to make this change, but I do not understand it. I am mirroring the hooks similar to this one, and they are all using `&`. Why is this hook different?

Reply:
The difference is you capture a copy or a reference, `&` is by reference, `=` is by copy.
So with the change, it means you cannot affect the original player object because you are accessing a copy of that player object. This effectively makes it read-only since none of the writes will affect the original object.

Reply:
> The difference is you capture a copy or a reference, `&` is by reference, `=` is by copy. So with the change, it means you cannot affect the original player object because you are accessing a copy of that player object. This effectively makes it read-only since none of the writes will affect the original object.
> 
> Whether or not this is intended is up to you to dispute with Winfidonarleyan 😄

Only copy `ptr` player, object player not copy

Reply:
https://en.cppreference.com/w/cpp/language/lambda `Lambda capture`